apps/impulse/lib/screens/onboarding_screen.dart

```diff
@@ -174,7 +174,7 @@ class _OnboardingScreenSate extends State<OnBoardingScreen> {
                     onTap: () {
                       Navigator.of(context).push(
                         MaterialPageRoute(builder: (context) {
-                          if (userNotifier.user.isEmpty) {
+                          if (userNotifier.user.id.isEmpty) {
```

Comment:
rather, use some method on the userNotifier to check this like

```suggestion
                          if (!userNotifier.hasUser ) {
```

and

```dart
bool get hasUser => !user.id.isEmpty
```